### PR TITLE
Allow insecure loads in requires_app_host's Info.plist

### PR DESCRIPTION
**Pull request by paulb777**
Allow insecure loads in requires_app_host's Info.plist so that tests that depend http loading will pass.

This is necessary for the GULNetworkTests in GoogleUtilities.podspec to pass when using test_spec's. See https://github.com/firebase/firebase-ios-sdk/pull/2908

More context in https://stackoverflow.com/a/31807139/556617

If this is an acceptable solution, I'll follow up with the Changelog and integration test updates.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Allow insecure loads in requires_app_host's Info.plist  
  [paulb777](https://github.com/paulb777)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb [hunk 1]
```diff
@@ -83,6 +83,12 @@ def install!
 
           private
 
+          ALLOW_INSECURE_LOADS_FOR_TESTS = {
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: Is this declared separate due to the fact that we want it to work on macOS as well?

**Comment by dnkoutso** [comment]
LGTM. yes would love tests fo this!

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb (hunk 1 above)
Comment (paulb777) [reply lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: Yes, and tvOS

**Comment by paulb777** [comment]
@dnkoutso Thanks for the quick review! I'll follow up this afternoon.

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb (hunk 1 above)
Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: maybe a more explicit variable name like `ADDITIONAL_INFO_PLIST_ENTRIES` (without the _IOS_)?

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb [hunk 2]
```diff
@@ -83,6 +83,12 @@ def install!
...
+            'NSAppTransportSecurity' => {
+              'NSAllowsArbitraryLoads' => true
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: shouldn't this be 'YES'`?

Comment (paulb777) [reply lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: It's a boolean

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb [hunk 3]
```diff
@@ -74,7 +74,7 @@ def install!
 
             Pod::Generator::AppTargetHelper.add_app_host_main_file(project, app_host_target, platform_name, @group, app_target_label) if add_main
             Pod::Generator::AppTargetHelper.add_launchscreen_storyboard(project, app_host_target, @group, deployment_target, app_target_label) if platform == :ios
-            additional_entries = platform == :ios ? ADDITIONAL_IOS_INFO_PLIST_ENTRIES : {}
+            additional_entries = ALLOW_INSECURE_LOADS_FOR_TESTS.merge(platform == :ios ? ADDITIONAL_IOS_INFO_PLIST_ENTRIES : {})
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: is there a test that covers the additional entries already? maybe integration spec? should be sufficient

**Comment by dnkoutso** [comment]
Just needs rubocop

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb (hunk 1 above)
Comment (paulb777) [reply lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: Done.

File: lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb (hunk 3 above)
Comment (paulb777) [reply lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb]: Yes the integration test covers it.

**Comment by paulb777** [comment]
Should be ready to merge. Thanks @dnkoutso!

**Pull request merged**